src/node_api_async.cc

```diff
@@ -0,0 +1,69 @@
+#include "node_api_async.h"
+#include "uv.h"
+
+typedef struct napi_work_impl__ {
+  uv_work_t* work;
+  void* data;
+  void (*execute)(void* data);
+  void (*complete)(void* data);
+  void (*destroy)(void* data);
+} napi_work_impl;
+
+napi_work napi_create_async_work() {
+  napi_work_impl* worker =
+      reinterpret_cast<napi_work_impl*>(malloc(sizeof(napi_work_impl)));
+  uv_work_t* req = reinterpret_cast<uv_work_t*>(malloc(sizeof(uv_work_t)));
+  req->data = worker;
+  worker->work = req;
+  return reinterpret_cast<napi_work>(worker);
```

Comment:
Should this clear the `execute/complete/destroy` fields by setting them to `nullptr`?